### ISSUE TITLE
Download button in a custom sidebar

### ISSUE THREAD

**Issue by m-fehr** [quarto-dev/quarto-cli#9388]
### Bug description

Using the code from here: https://github.com/quarto-dev/quarto-cli/discussions/4235 I tried to make a customized sidebar.
When I added the "_downloads: [pdf]_" line nothing changed. Using the automatically genereated sidebar I get the download button.

### Steps to reproduce

I made a MWE using a freshly setup quarto book project which can be found here: 
https://github.com/m-fehr/test_custom_sidebar_download

The problematic code I think, is this part in the yaml file

```qmd
  sidebar: 
    - title: "Test"
      downloads: [pdf]
      tools:
      - icon: github
        href: https://github.com/
      
      contents:
      - section: Introduction
        contents:
        - href: index.qmd
          text: Intro
```

### Expected behavior

I expect so see a download button next to the github symbol.

### Actual behavior

There is nothing.

### Your environment

- IDE: VSCode Version: 1.88.0
- OS: MacOS Sonoma 14.3

### Quarto check output

```bash
Quarto 1.4.553
[✓] Checking versions of quarto binary dependencies...
      Pandoc version 3.1.11: OK
      Dart Sass version 1.69.5: OK
      Deno version 1.37.2: OK
[✓] Checking versions of quarto dependencies......OK
[✓] Checking Quarto installation......OK
      Version: 1.4.553
      Path: /Applications/quarto/bin

[✓] Checking tools....................OK
      TinyTeX: v2024.02
      Chromium: 869685

[✓] Checking LaTeX....................OK
      Using: TinyTex
      Path: /Users/mfehr/Library/TinyTeX/bin/universal-darwin
      Version: 2023

[✓] Checking basic markdown render....OK

[✓] Checking Python 3 installation....OK
      Version: 3.12.2
      Path: /opt/homebrew/opt/python@3.12/bin/python3.12
      Jupyter: (None)

      Jupyter is not available in this Python installation.
      Install with python3 -m pip install jupyter

[✓] Checking R installation...........OK
      Version: 4.3.3
      Path: /opt/homebrew/Cellar/r/4.3.3/lib/R
      LibPaths:
        - /opt/homebrew/lib/R/4.3/site-library
        - /opt/homebrew/Cellar/r/4.3.3/lib/R/library
      knitr: 1.45
      rmarkdown: 2.26

[✓] Checking Knitr engine render......OK
```

**Comment by mcanouil** [comment]
Thanks for the report.
Could you fix the formatting of your post?

**Comment by cscheid** [comment]
Sorry - why do you expect that syntax to work?

**Comment by m-fehr** [comment]
In the example provided here, the "search: true " key is also set at this point in the yaml file. But even if I do put it in in the normal way of 
```
book:
  title: "test_custom_sidebar_download"
  author: "Marc Fehr"
  downloads: [pdf]
```
no download button appears.

**Comment by mcanouil** [comment]
Be sure to double check your project.

1. `quarto create project book Book`
2. Adding `  downloads: [pdf]` to `_quarto.yml`.

<img width="1624" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/5b8f66a8-d002-475c-8661-3fb379a15de6">


Are you sure about the link you shared? There absolutely no examples with `downloads` (also no examples involving Book project type).

**Comment by m-fehr** [comment]
My aim was to have a sidebar with multiple levels of items (more than one), which I realize now is not clear in my example. Sorry for that! That's why I used the above mentioned example to build a multilevel sidebar. Now I want to add the download button. But even doing:
```
project:
  type: book

book:
  title: "test_custom_sidebar_download"
  author: "Marc Fehr"
  downloads: [pdf]
  date: "4/16/2024"
  chapters:
    - index.qmd
    - intro.qmd
    - summary.qmd
    - references.qmd
  sidebar: 
    - title: "Test"
      contents:
      - section: Introduction
        contents:
        - href: index.qmd
          text: Intro
```
gives me the sidebar I need (here without multiple levels to keep it short) but no download button.

Is there a way to combine the navigation described here: https://quarto.org/docs/websites/website-navigation.html#side-navigation with a download button. This should work according to https://quarto.org/docs/books/book-output.html#sidebar-options

Sorry again for the confusion!

**Comment by m-fehr** [comment]
My goal was to have a sidebar like here: https://firedynamics.github.io/LectureFireSimulation/content/overview/01_overview.html

Since my files contain their respective section only I have to use include in the chapters to import them. And because the part/chapter solution you mentioned in your last comment only displays part and chapter and not section in the sidebar I used the code snippets described above.

Using the above code with the title key, I was able to define my own sidebar with links to the sections inside the chapters inside the parts, but I now understand this was non-intended behaviour. I will stick to what's documented in the future. Again sorry for the unnecessary issue.
